pkg/nsx/services/realizestate/realize_state.go

```diff
@@ -88,7 +88,8 @@ func (service *RealizeStateService) GetPolicyTier1UplinkPortIP(intentPath string
 
 	for _, result := range results.Results {
 		extendAttributes := result.ExtendedAttributes
-		if len(extendAttributes) == 0 || len(result.IntentPaths) != 1 || (result.EntityType != nil && *result.EntityType != "RealizedLogicalRouterPort") {
+		if len(extendAttributes) == 0 || len(result.IntentPaths) != 1 || *result.Id != "gateway-interface" ||
+			(result.EntityType != nil && *result.EntityType != "RealizedLogicalRouterPort") {
```

Comment:
Can we use ID to get the realized state directly?

Reply:
OK, close this PR, as it is fixed at https://github.com/vmware-tanzu/nsx-operator/pull/1082#issue-3019217424